src/app/tests/TestReportScheduler.cpp

```diff
@@ -387,7 +457,351 @@ class TestReportScheduler
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler3));
+        readHandler1->mObserver->OnBecameReportable(readHandler1);
+        readHandler2->mObserver->OnBecameReportable(readHandler2);
+        readHandler3->mObserver->OnBecameReportable(readHandler3);
+        // Engine run should happen here and send all reports
+        readHandler1->mObserver->OnSubscriptionAction(readHandler1);
+        readHandler2->mObserver->OnSubscriptionAction(readHandler2);
+        readHandler3->mObserver->OnSubscriptionAction(readHandler3);
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler1));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler2));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler3));
+        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node1->GetMaxTimestamp());
+        NL_TEST_ASSERT(aSuite, node1->GetSyncTimestamp() == node1->GetMaxTimestamp());
+        NL_TEST_ASSERT(aSuite, node2->GetSyncTimestamp() == node1->GetMaxTimestamp());
+
+        // Now simulate a new readHandler being added with a max forcing a conflict
+        ReadHandler * readHandler4 =
+            readHandlerPool.CreateObject(nullCallback, exchangeCtx, ReadHandler::InteractionType::Subscribe);
+        NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler4->SetMaxReportingInterval(1));
+        NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler4->SetMinReportingIntervalForTests(0));
+        readHandler4->MoveToState(ReadHandler::HandlerState::GeneratingReports);
+        readHandler4->SetObserver(&syncScheduler);
+        readHandler4->mObserver->OnReadHandlerCreated(readHandler4);
+
+        // Confirm all handler are currently registered in the scheduler
+        NL_TEST_ASSERT(aSuite, syncScheduler.GetNumReadHandlers() == 4);
+        ReadHandlerNode * node4 = syncScheduler.FindReadHandlerNode(readHandler4);
+
+        // Confirm next report is scheduled on the max timestamp of readHandler4 and other handlers be synced
+        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node4->GetMaxTimestamp());
+        NL_TEST_ASSERT(aSuite, node1->GetSyncTimestamp() == node4->GetMaxTimestamp());
+        NL_TEST_ASSERT(aSuite, node2->GetSyncTimestamp() == node4->GetMaxTimestamp());
+        NL_TEST_ASSERT(aSuite, node3->GetSyncTimestamp() == node1->GetMaxTimestamp());
+
+        sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(1100));
+
+        // Confirm readHandler4, 1 and 1 are reportable
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler3));
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler4));
```

Comment:
```suggestion
        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler4));
        
        // ReadHandler3 is not reportable
        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler3));
```

this makes the block clearer